File: LC/python/Sqrt(x).py
class Solution:
    # @param x, an integer
    # @return an integer
    def sqrt(self, x):
        if x < 2:
            return x
        
        left = 0 
        right = x/2
        
        if right**2 == x:
            return right
        
        while left <= right:
            mid = (left+right)/2
            if mid**2 == x:
                return mid
            elif mid**2 < x:
                left = mid+1
            else:
                right = mid-1
                
        return right
                

File: LC/java/Sqrt(x).java
public class Solution {
    public int sqrt(int x) {
        long l = 0, r = x;
        
        while(l<=r){
            long mid = l+(r-l)/2;
            if(mid*mid > x){
                r=mid-1;
            }
            else{
                l=mid+1;
            }
        }
        return (int)r;
    }
}